frontends.tensorflow: add support for full function

hmahmood24 opened the pull request:
Close #6968

ivy-seed commented:
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

Ishticode reviewed (changes requested):
Hi @hmahmood24 Thank you for the PR.
Could you remove `@handle_cmd_line_args` and update `@given` to `@handle_frontend_test`... You will be able to see this in other tests within frontend tests. This is due to recent change and the documentation still requires updating. Apologies about that. Please let me know if something is unclear. After you've done this and resolved the conflicts shown below please request a review and we will try to merge this. Thanks for you pattience and feel free to ask anything. Thank you.

hmahmood24 commented:
Hi @Ishticode,

Thanks for the response. I have refactored my tests to reflect these recent changes in the frontend tests. However, when I try to run my tests, I get the following error: `frontend_ret = frontend_fw.__dict__[fn_name](*args_frontend, **kwargs_frontend). KeyError: 'full'` in `ivy/ivy_tests/test_ivy/helpers/function_testing.py:614`

Upon some debugging and printing, the value for `frontend_fw` comes out to be `<module 'tensorflow' from '/home/haris/ivy/lib/python3.8/site-packages/tensorflow/__init__.py'>` and for `fn_name` comes out to be `full`. When I tried to access `tensorflow.full` it throws the following error: `module 'tensorflow' has no attribute 'full'`. 

The function I'm trying to implement is actually located in `tensorflow.experimental.numpy.full` in the original tensorflow module. This means that the `fn_tree` argument that I have to provide with my test cannot be `tensorflow.full` because it throws an error with that as shown above. It also cannot be `tensorflow.experimental.numpy.full` because this also throws an error when locating in the `ivy.functional.frontends.tensorflow` module.

Upon going through Ivy's [docs](https://lets-unify.ai/ivy/deep_dive/ivy_frontends_tests.html#ivy-frontend-tests), I found this:

> Tensorflow’s version of full() is named fill() therefore we specify the fn_tree argument to be "fill"

and so following that, I changed my function signature from that of `full(shape, fill_value, *, dtype=None)` to `fill(dims, value, *, name=None)`, both using `ivy.full(...)` under the hood since there is no explicit implementation of `ivy.fill` in the codebase. However my tests still fail with the following error: `AssertionError: the return with a tensorflow backend produced data type of float32, while the return with a torch backend returned a data type of float64.`. The problem now is probably that `full(...)` uses a `dtype` parameter when creating an array whereas `fill(...)` does not have any such parameter. I still have committed my code into this PR since I'm out of ideas as to how can I go about fixing this. Can you please guide me moving forward from here?

hmahmood24 commented:
@Ishticode Waiting to hear from you so we can resolve this PR asap.

Ishticode reviewed (changes requested):
Hi @hmahmood24 Thanks for the review request. Could you kindly exclude any changes to `ivy_tests/test_array_api`. Also, the `name` arg in tensorflow's fill function is not kw-only so we wouldn't require `*`.  Let me know if something is unclear. Thank you 🙂

Ishticode reviewed (approved):
@hmahmood24 
This looks great. Just had to make minor changes in naming the pr/issue title and test name. The CI had a lot of logs without any result for test_tensorflow_fill so I had to check locally and it runs fine. Thanks for your contribution and your patience. Much appreciated.
